ui/app/adapters/mfa-method.js

```diff
@@ -7,19 +7,30 @@ export default class MfaMethodAdapter extends ApplicationAdapter {
     return 'identity/mfa/method';
   }
 
-  queryRecord(store, type, query) {
-    const { id } = query;
-    if (!id) {
-      throw new Error('MFA method ID is required to fetch the details.');
-    }
-    const url = this.urlForQuery(query, type.modelName);
-    return this.ajax(url, 'POST', {
-      data: {
-        id,
-      },
+  createOrUpdate(store, type, snapshot) {
+    const data = store.serializerFor(type.modelName).serialize(snapshot);
+    const { id } = snapshot;
+    return this.ajax(this.buildURL(type.modelName, id, snapshot, 'POST'), 'POST', {
+      data,
+    }).then(() => {
+      // TODO: Check how 204's are handled by ember
+      return {
+        data: {
+          id,
+          ...data,
+        },
+      };
```

Comment:
I'm not sure this will work as expected. In the case of creating a new record the id from the snapshot will be undefined but the id is returned in the response, whereas updating returns no content. Also, I don't think we want to return the data under the data key.

Reply:
Following up after looking at the serializer. The `normalizeItems` method expects a data object in the payload so disregard that part of the comment.